charts/admission-controller/templates/_helpers.tpl

```diff
@@ -110,26 +110,47 @@ If release name contains chart name it will be used as a full name.
     {{- end -}}
 {{- end -}}
 
+{{/*
+Validate Sysdig API Endpoint
+*/}}
+{{- define "chart.validateSysdigApiEndpoint" -}}
+    {{- if or (hasPrefix "http://" .Values.sysdig.apiEndpoint) (hasPrefix "https://" .Values.sysdig.apiEndpoint) -}}
+        {{- fail "Sysdig API endpoint should not contain http:// or https:// prefix" -}}
+    {{- end -}}
+{{- end -}}
+
 {{/*
 Determine Secure endpoint based on provided region or .Values.sysdig.apiEndpoint
 */}}
 {{- define "admissionController.apiEndpoint" -}}
-    {{- if (or .Values.sysdig.apiEndpoint (eq .Values.global.sysdig.region "custom"))  -}}
-        {{- required "A valid Sysdig API endpoint (.sysdig.apiEndpoint) is required" .Values.sysdig.apiEndpoint -}}
-    {{- else if hasKey ((include "sysdig.regions" .) | fromYaml) .Values.global.sysdig.region }}
-        {{- include "sysdig.secureApiEndpoint" . }}
+    {{ include "chart.validateSysdigApiEndpoint" . }}
+    {{- if eq .Values.global.sysdig.region "custom" -}}
+        {{- $apiEndpoint := coalesce .Values.sysdig.url .Values.sysdig.apiEndpoint -}}
+        {{- required "A valid Sysdig API endpoint (.apiEndpoint or .url) is required for custom region" $apiEndpoint -}}
+    {{- else }}
+        {{- if .Values.sysdig.apiEndpoint -}}
```

Comment:
Since `.Values.sysdig.apiEndpoint` should be used only if `custom` region has been chosen, i'm not sure why we are still checking it here 🤔

Reply:
I don't know if I properly understood your comment, but **apiEndpoint** isn't only used if **custom** region has been chosen. 

If **custom** region has been chosen we require either **url** or **apiEndpoint**, but for non-custom regions we still provide the option to receive **apiEndpoint**, then in the last case we fallback to default endpoint from _regions.tpl.